tests/unit/helpers/sl/synchronous/requires-test.js

```diff
@@ -12,283 +12,165 @@ test( 'it exists', function( assert ) {
-    } catch( error ) {
-        assertionThrown = true;
-    }
+    testPropertyOne.set( 'parameter', [] );
 
-    assert.ok(
-        assertionThrown,
+    assert.throws(
+        callRequires,
         'First parameter was an array'
     );
 
     // Object
-    assertionThrown = false;
+    testPropertyOne.set( 'parameter', {} );
 
-    try {
-        requires(
-            {},
-            false
-        );
-    } catch( error ) {
-        assertionThrown = true;
-    }
-
-    assert.ok(
-        assertionThrown,
+    assert.throws(
+        callRequires,
         'First parameter was an object'
     );
 
     // Function
-    assertionThrown = false;
+    testPropertyOne.set( 'parameter', function() {} );
 
-    try {
-        requires(
-            function() {},
-            false
-        );
-    } catch( error ) {
-        assertionThrown = true;
-    }
-
-    assert.ok(
-        assertionThrown,
+    assert.throws(
+        callRequires,
         'First parameter was a function'
     );
 
     // Undefined
-    assertionThrown = false;
+    testPropertyOne.set( 'parameter', undefined );
 
-    try {
-        requires(
-            undefined,
-            false
-        );
-    } catch( error ) {
-        assertionThrown = true;
-    }
-
-    assert.ok(
-        assertionThrown,
+    assert.throws(
+        callRequires,
         'First parameter was undefined'
     );
 
     // Boolean
-    assertionThrown = false;
+    testPropertyOne.set( 'parameter', false );
 
-    try {
-        requires(
-            true,
-            false
-        );
-    } catch( error ) {
-        assertionThrown = true;
-    }
-
-    assert.ok(
-        assertionThrown,
+    assert.throws(
+        callRequires,
         'First parameter was a boolean'
     );
 
-    // Null
-    assertionThrown = false;
-
-    try {
-        requires(
-            null,
-            false
-        );
-    } catch( error ) {
-        assertionThrown = true;
-    }
+    // Function, Array
+    testPropertyOne.set( 'parameter', function() {} );
+    testPropertyTwo.set( 'parameter', [] );
 
     assert.ok(
-        assertionThrown,
-        'First parameter was null'
+        callRequires(),
+        'First parameter was a function and second parameter is an array'
     );
-
 });
 
 test( 'Second argument must be an array', function( assert ) {
+    const testPropertyOne = Ember.Object.create({
+        parameter: undefined
+    });
 
-    // Number
-    let assertionThrown = false;
+    const testPropertyTwo = Ember.Object.create({
+        parameter: undefined
+    });
 
-    try {
-        requires(
-            12,
-            12
-        );
-    } catch( error ) {
-        assertionThrown = true;
-    }
+    const callRequires = () => requires( testPropertyOne.parameter, testPropertyTwo.parameter );
 
-    assert.ok(
-        assertionThrown,
+    // Number
+    testPropertyOne.set( 'parameter', 123 );
+    testPropertyTwo.set( 'parameter', 123 );
+
+    assert.throws(
```

Comment:
Pretty sure this assertion is passing because `callRequires` throws an error not because the second parameter is a Number and not an Array, but because the first parameter is not a Function.  Confirm that this is the case - think it has been incorrect for a while, and not anything introduced in this PR specifically, just now became evident.

This is the case for all the assertions in this test.